uint32 MapWidth
uint32 MapHeight
uint8 MapData